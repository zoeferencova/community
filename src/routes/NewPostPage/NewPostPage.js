import React, { Component } from "react";
import { Link } from "react-router-dom";
import CommUnityContext from "../../contexts/context";
import styles from "./NewPostPage.module.css";
import UserDataService from "../../services/user-data-service";

export default class NewPostPage extends Component {
    static contextType = CommUnityContext;
    
    handleSubmit(e) {
        e.preventDefault();

        const post_type = this.props.match.params.type;
        let category_ids = [], option;
        for (let i = 0; i < e.target.categories.length; i++) {
          option = e.target.categories[i];
    
          if (option.selected) {
            category_ids.push(i+1);
          }
        }
        
        const post = { post_type, category_ids }

        if (e.target.description.value) {
            post.description = e.target.description.value;
        }

        if (post_type === "request" && e.target.urgency.value) {
            post.urgency = (e.target.urgency.value).toLowerCase();
        }

        UserDataService.postPost(post)
            .then(post => {
                this.context.addNewPost(post)
                this.props.history.push("/home")
            })
    }
    
    render() {
        const type = this.props.match.params.type;
        return (   
            <main>
                <h3>Create a new {type}</h3>
                <form className={styles.form} onSubmit={e => this.handleSubmit(e)}>
                    <div>
                        <label className={styles.label} htmlFor="categories">{type === "offer" ? "What can you help with?": "What do you need help with?"}</label>
                        <select className={styles.select} id="categories" multiple>
                            <option value="Picking up supplies">Picking up supplies</option>
                            <option value="Running errands">Running errands</option>
                            <option value="Phone call">Phone call</option>
                            <option value="Online chat">Online chat</option>
                            <option value="Dog walking">Dog walking</option>
                            <option value="Other">Other</option>
                        </select>
                    </div>
                    {type === "request" && <div>
                        <label className={styles.label} htmlFor="urgency">Urgency</label>
                        <select className={styles.select} id="urgency">
                            <option value="Low">Low</option>
                            <option value="Medium">Medium</option>
                            <option value="Urgent">Urgent</option>
                        </select>
                    </div>}
                    <div>
                        <label className={styles.label} htmlFor="description">Description (optional)</label>
                        <textarea className={styles.textarea} name="description" id="description"></textarea>
                    </div>
                    <Link to="/home"><button>Cancel</button></Link>
                    <button type="submit">Submit</button>
                </form>
            </main>
        )
    }
}