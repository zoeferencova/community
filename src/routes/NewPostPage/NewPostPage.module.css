.form {
    width: 75%;
    margin-bottom: 20px;
}

.form h3 {
    font-size: 18px;
    font-weight: 500;
    margin-bottom: 30px;
}

.textarea {
    height: 100px !important;
}

.label {
    margin-bottom: 10px;
}

.buttonContainer {
    display: flex;
    justify-content: space-between;
}

.error {
    text-align: left;
    font-size: 12px;
    color: #a34343;
    margin: -15px 0px 15px 0px;
}