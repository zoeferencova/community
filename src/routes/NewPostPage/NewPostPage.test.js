import React from "react";
import { render } from "enzyme";
import NewPostPage from "./NewPostPage";
import { MemoryRouter, Route } from "react-router-dom";
import CommUnityContext from "../../contexts/context";

import mockContext from "../../contexts/mock-context";

describe("NewPostPage", () => {
    it("renders NewPostPage component given mock context data", () => {
        const wrapper = render(
            <CommUnityContext.Provider value={mockContext}>
                <MemoryRouter initialEntries={[`/new-post/request`]}>
                    <Route path="/new-post/:type">
                        <NewPostPage />
                    </Route>
                </MemoryRouter>
            </CommUnityContext.Provider>,
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});