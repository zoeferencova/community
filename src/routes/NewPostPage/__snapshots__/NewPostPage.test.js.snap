// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPostPage renders NewPostPage component given mock context data 1`] = `
<main
  class="main"
>
  <h3>
    Create a new request 
    <i
      class="fas fa-hand-paper hand"
    />
  </h3>
  <form
    class="form"
  >
    <div>
      <label
        class="label label"
        for="categories"
      >
        What do you need help with?
      </label>
      <select
        id="categories"
        multiple=""
        name="categories"
      >
        <option
          class="supplies"
          value="Picking up supplies"
        >
          Picking up supplies
        </option>
        <option
          id="errands"
          value="Running errands"
        >
          Running errands
        </option>
        <option
          class="phone"
          value="Phone call"
        >
          Phone call
        </option>
        <option
          class="online"
          value="Online chat"
        >
          Online chat
        </option>
        <option
          class="dog"
          value="Dog walking"
        >
          Dog walking
        </option>
        <option
          class="other"
          value="Other"
        >
          Other
        </option>
      </select>
    </div>
    <div>
      <label
        class="label label"
        for="urgency"
      >
        Urgency
      </label>
      <select
        id="urgency"
        required=""
      >
        <option
          value="Low"
        >
          Low
        </option>
        <option
          value="Medium"
        >
          Medium
        </option>
        <option
          value="High"
        >
          High
        </option>
      </select>
    </div>
    <div>
      <label
        class="label label"
        for="description"
      >
        Description (optional)
      </label>
      <textarea
        class="input textarea"
        id="description"
        name="description"
      />
    </div>
    <div
      class="buttonContainer"
    >
      <button
        class="button "
        type="button"
      >
        Cancel
      </button>
      <button
        class="buttonDark "
        loading="false"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</main>
`;
